### ISSUE TITLE
warning when searching without authentication and/or not getting results?

### ISSUE THREAD

**Issue by aappling-usgs** [USGS-R/sbtools#158]
A philosophical question: Can and should `sbtools` usefully steer users to more successful querying by noticing when authentication is missing and/or when 0 search results are returned, then providing a warning message suggesting authentication?

(I raised this question as part of an earlier issue (#132) - moving it to its own issue here to make sure it gets some discussion.)

Example: Running this search without authentication returns 0 items with no errors or warnings.
```r
> library(sbtools)
> run_items <- query_item_identifier(scheme = 'mda_streams', type = 'metab_run', limit = 10000)
> dim(run_items)
[1] 0 0
```
Whereas running the search after authentication returns 14 items (provided your account has access to these items).
```r
> authenticate_sb()
> run_items <- query_item_identifier(scheme = 'mda_streams', type = 'metab_run', limit = 10000)
> dim(run_items)
[1] 14  2
```
Also, if you wait about an hour or more, your authentication silently expires. Within the same session as above, then, it's possible to go back to 0 items returned without warning:
```r
> Sys.time() # authenticated right now, but wait a while
[1] "2016-01-18 12:39:14 CST"
> Sys.time() # OK, been waiting a while
[1] "2016-01-18 13:53:40 CST"
> run_items <- query_item_identifier(scheme = 'mda_streams', type = 'metab_run', limit = 10000)
> dim(run_items)
[1] 0 0
```

The current behavior is OK - many queries can and should be run without authentication. But suppose you're using `sbtools` and were recently authenticated in your R session. Then you run some query and are suddenly getting back fewer / no results - do you (1) brilliantly realize that your session has timed out and go fix that first, (2) not notice the change because this query was buried in a long script and returned no errors or warnings, or (3) notice and panic because SB seems to be malfunctioning and has precedent for malfunctioning at inconvenient times, and you really needed to get that report out today...? I argue that not every user will go straight to the correct response every time. 

When would response 1 be the correct one, and is there any way for us to know from the `sbtools` side? We'll never know for sure, but it seems most likely when (a) the user is not authenticated, (b) the session has expired, and/or (c) the query has returned 0 results. If we returned a warning message under these conditions, we might help some users through some frustrating and hard-to-replicate problems.

**Comment by lawinslow** [comment]
I've tossed up a simple change that would in general address this. Pretty much the point you're making is that if the session expires for any reason, the user may not know and have unpredictable or bizarre results. To deal with this in a general sense, if you're relying on the persisted session (via `current_session`), and that session is expired, then a warning would be helpful. I've added this warning.
